src/org/labkey/test/components/react/dialogs/BulkCreateSamplesDialog.java

```diff
@@ -0,0 +1,160 @@
+package org.labkey.test.components.react.dialogs;
+
+import org.labkey.test.BootstrapLocators;
+import org.labkey.test.Locator;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.labkey.test.components.glassLibrary.components.FilteringReactSelect;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_JAVASCRIPT;
+
+public class BulkCreateSamplesDialog extends ModalDialog
+{
+
+    public BulkCreateSamplesDialog(WebDriver driver)
+    {
+        this(new ModalDialogFinder(driver).withTitle("Bulk Creation of Samples"));
+    }
+
+    private BulkCreateSamplesDialog(ModalDialogFinder finder)
+    {
+        super(finder);
+    }
```

Comment:
Take an `InsertEntitiesPanel` in the constructor. That way this component could wait for the list to update in `clickAddRows`.
```suggestion
    private final InsertEntitiesPanel _panel;

    public BulkCreateSamplesDialog(InsertEntitiesPanel panel)
    {
        super(new ModalDialogFinder(panel.getDriver()).withTitle("Bulk Creation of Samples"));
        _panel = panel;
    }
```
`InsertEntitiesPanel.getDriver()` needs to be public for this to work (I really wish I'd made `WebDriverComponent.getDriver` public in the first place 🤦)

Reply:
BCSD now takes an InsertEntitesPanel in its constructor, getDriver() is now public on InsertEntitiesPanel